internal/api/v1/organizations.go

```diff
@@ -162,7 +162,50 @@ func deleteApps(cluster *kubernetes.Cluster, gitea *gitea.Client, org string) er
 		return err
 	}
 
-	// create a buffer, so there is no more than 'maxConcurrent' goroutines running at the same time
+	// Operation of the concurrent code below:
```

Comment:
I think any method that needs so much documentation is probably a method that should be re-written in a simpler way. On the other hand, I can't think of a simpler way right now so I'm ok to stick with the docs.

Reply:
I suspect that even "simple" concurrent code, like we have here, always will at least this much documentation.